### DISCUSSION TITLE
Command + W Key Binding

### DISCUSSION THREAD

**Discussion by 201flaviosilva** [gitpod-io/openvscode-server#322] [Q&A] [answered]
Hey there,

This question might seem a bit silly and unnecessary, but is there any wood to force key combinations mostly to `Command+W`, it happens with some frequency.

I want to close a file I'm not using any more, and sometimes I close the whole tab 🥲. As this is a VS Code shortcut, I thought it would make sense to have it here too.

Thanks any help :)

(Sorry if I should ask this in the [gitpod](https://github.com/gitpod-io/gitpod) repository)

**Comment by filiptronicek** [comment, marked as answer]
Hey there @201flaviosilva, this is not possible under normal circumstances in the browser, there is no way a website can capture these special keybinds like <kbd>Ctrl + N</kbd> or <kbd>Ctrl + W</kbd>. This could be only possible using the [Keyboard lock API](https://developer.mozilla.org/en-US/docs/Web/API/Keyboard/lock), which not a good option now for two reasons:
1. It is still experimental, no support for most browsers, [basically just Chrome](https://caniuse.com/mdn-api_keyboard_lock).
2. The browser tab has to be in full screen mode which is most of the time not desirable, but I'd love to discuss this with the VS Code team @ Microsoft.

**Reply by filiptronicek** [reply]
But I want to point out I really miss this too, so many of us do here at Gitpod :)

**Reply by 201flaviosilva** [reply]
Ohhh, I didn't know about all these issue, but very grateful for everything anyway, and nice work :)

**Reply by 201flaviosilva** [reply]
Maybe one day later, don't close the tab any more 🥲 😅

**Reply by rutsky** [reply]
If you sticking with Chrome, you can run OpenVSCode window as an "application" (3 dots -> `More tools` -> `Create shortcut...` -> `Open as window` checkbox: https://askubuntu.com/a/910516), which makes most of the shortcuts work on Linux, including Ctrl+W, Ctrl+N, etc (not sure about Mac since you asking about *Command* + W shortcut).
